flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobAlreadyDoneException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.JobException;
+
+/** Exception indicating that a Flink job failed in finishing phase. */
+class JobAlreadyDoneException extends JobException {
+    private static final long serialVersionUID = 5906282500322681417L;
+
+    JobAlreadyDoneException(JobID jobId) {
+        super(
+                "The job ("
+                        + jobId
+                        + ") has already terminated, but we cannot tell the final "
+                        + "state and also not serve the jobResult for some unexpected reason.");
```

Comment:
```suggestion
                        + "state and also not serve the job result. This can happen if Flink fails over shortly after finishing a job and before reporting the result. Please check the output of your job to see whether the execution was successful or not.");
```

Reply:
Done